### ISSUE TITLE
[BUG] Invalid YAML in repo-metadata.yaml silently defaults to internal tier

### ISSUE THREAD

**Issue by chrismlittle123** [chrismlittle123/check-my-toolkit#156]
## Description

When `repo-metadata.yaml` contains invalid YAML, tier validation silently defaults to "internal" without any warning about the parse error. This makes debugging difficult since the user has no indication their metadata file is broken.

## Steps to Reproduce

1. Create `check.toml`:
```toml
[extends]
registry = "github:example/standards"
rulesets = ["base-production"]
```

2. Create `repo-metadata.yaml` with invalid YAML:
```yaml
tier: production
  bad: yaml: here
```

3. Run:
```bash
cm validate tier
```

## Expected Result

A warning or error indicating the YAML parse failed:
```
⚠ Warning: Failed to parse repo-metadata.yaml: <parse error details>
  Defaulting to tier: internal

✗ Tier validation failed
  Tier: internal (source: default - parse error)
  ...
```

## Actual Result

```
✗ Tier validation failed
  Tier: internal (source: default)
  Expected pattern: *-internal
  Rulesets: [base-production]
  Error: No ruleset matching pattern '*-internal' found. Rulesets: [base-production]
```

No indication that:
- The YAML file exists but failed to parse
- The "default" source is due to a parse error vs missing file

## Comparison

| Scenario | Result |
|----------|--------|
| Valid YAML | ✅ `Tier: production (source: repo-metadata.yaml)` |
| Invalid YAML | ❌ `Tier: internal (source: default)` - no warning |
| Missing file | ❌ `Tier: internal (source: default)` - same as invalid |

## Impact

Users may spend time debugging ruleset mismatches when the actual problem is a syntax error in their `repo-metadata.yaml`. The silent fallback masks the root cause.

## Suggested Fix

In `loadRepoMetadata()` function, catch the YAML parse error and either:
1. Log a warning to stderr
2. Include parse error info in the result
3. Return a distinct source like `"default (parse error)"` or `"error"`

## Environment

- check-my-toolkit version: 1.4.0
- Node.js version: 25.2.1
- Platform: macOS (Darwin 25.2.0)
